core/tee/tee_svc_storage.c

```diff
@@ -978,6 +978,7 @@ TEE_Result tee_svc_storage_start_enum(uint32_t obj_enum, uint32_t storage_id)
 			if (res != TEE_SUCCESS)
 				goto exit;
 			res = tee_obj_verify(sess, o);
+			free(o->pobj->obj_id);
```

Comment:
The need for this free is a bit obscure, please add a comment explaining why it's needed here.